README.md

```diff
@@ -103,74 +104,123 @@ install all the `*.opam` packages stored in this directory.
 *Optional* The version of OCaml. Default `"minimal"`.
 Among `"minimal"`, `"4.07-flambda"`, `"4.09-flambda"`.
 
+#### `before_install`
+
+*Optional* The bash snippet to run before `install`. Default `""` (empty string).
+
+See [`custom_script`](#custom_script) for more details.
+
+#### `install`
+
+*Optional* The bash snippet to install the `opam` `PACKAGE` dependencies.
+
+Default:
+
+```
+opam pin add -n -y -k path $PACKAGE $WORKDIR
+opam update -y
+opam install -y -j 2 $PACKAGE --deps-only
+```
+
+See [`custom_script`](#custom_script) for more details.
+
+#### `after_install`
+
+*Optional* The bash snippet to run after `install` (if successful).
+
+Default:
+
+```
+opam list
+```
+
+See [`custom_script`](#custom_script) for more details.
+
+#### `before_script`
+
+*Optional* The bash snippet to run before `script`. Default `""` (empty string).
+
+See [`custom_script`](#custom_script) for more details.
+
+#### `script`
+
+*Optional* The bash snippet to install the `opam` `PACKAGE`.
+
+Default:
+
+```
+opam install -y -v -j 2 $PACKAGE
+opam list
+```
+
+See [`custom_script`](#custom_script) for more details.
+
+#### `after_script`
+
+*Optional* The bash snippet to run after `script` (if successful). Default `""` (empty string).
+
+See [`custom_script`](#custom_script) for more details.
+
+#### `uninstall`
+
+*Optional* The bash snippet to uninstall the `opam` `PACKAGE`.
+
+Default:
+
+```
+opam remove $PACKAGE
+```
+
+See [`custom_script`](#custom_script) for more details.
+
 #### `custom_script`
 
-*Optional* The main script run in the container; may be overridden. Default:
+*Optional* The main script run in the container; may be overridden; but overriding more specific parts of the script is preferred. Default:
 
-```bash
+```
 startGroup Print opam config
   opam config list; opam repo list; opam list
```

Comment:
What about making this the default `before_install`?

Reply:
@Zimmi48 on the one hand, this will force users interested in a custom `before_install` code *and* in `opam config list` etc. to re-copy it; but on the other hand, your suggestion is important for users that don't use a Docker-Coq/opam image.
So yes, that change looks sensible.